field/internal/templates/element/temp_for_hash.go

```diff
@@ -0,0 +1,39 @@
+package element
+
+const TempForHash = `
+
+// Sgn0 is an algebraic substitute for the notion of sign in ordered fields
+// Namely, every non-zero quadratic residue in a finite field of characteristic =/= 2 has exactly two square roots, one of each sign
+// Taken from https://datatracker.ietf.org/doc/draft-irtf-cfrg-hash-to-curve/ section 4.1
+// The sign of an element is not obviously related to that of its Montgomery form
+func (z *{{.ElementName}}) Sgn0() bool {
+	nonMont := *z
+	nonMont.FromMont()
+
+	return nonMont[0]%2 == 1
+}
+
+func (z *{{.ElementName}}) SetHex(hex string) {
```

Comment:
remove this function, Element.SetString() works with both hex (prefixed by 0x) and decimal strings, no need to add this extra API.